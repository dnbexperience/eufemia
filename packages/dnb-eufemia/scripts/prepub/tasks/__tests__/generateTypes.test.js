/**
 * Scripts test
 *
 */

import nodePath from 'path'
import { createTypes } from '../generateTypes'

describe('generateTypes', () => {
  it('has to match PrimaryComponent snapshot', async () => {
    const docsDir = nodePath.resolve(
      __dirname,
      '../generateTypes/__tests__',
      '__mocks__'
    )
    const file = nodePath.resolve(docsDir, 'PrimaryComponent.js')
    const [{ destFile, definitionContent }] = await createTypes([file], {
      isTest: true,
      docsDir,
      findFiles: ['PrimaryComponent.md']
    })

    expect(destFile).toContain('__mocks__/PrimaryComponent.d.ts')
    expect(definitionContent).toMatchSnapshot()
  })

  it('has to match SecondaryComponent snapshot', async () => {
    const docsDir = nodePath.resolve(
      __dirname,
      '../generateTypes/__tests__',
      '__mocks__'
    )
    const file = nodePath.resolve(docsDir, 'SecondaryComponent.js')
    const [{ destFile, definitionContent }] = await createTypes([file], {
      isTest: true,
      docsDir,
      findFiles: ['SecondaryComponent.md']
    })

    expect(destFile).toContain('__mocks__/SecondaryComponent.d.ts')
    expect(definitionContent).toMatchSnapshot()
  })
})
