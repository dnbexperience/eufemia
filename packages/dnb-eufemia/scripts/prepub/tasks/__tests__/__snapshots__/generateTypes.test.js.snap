// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTypes has to match PrimaryComponent snapshot 1`] = `
"import * as React from 'react';
export interface PrimaryComponentSpacing {
  /**
   * Text \`margin-top\`.
   */
  top?: string | number | boolean;

  /**
   * Text \`margin-right\`.
   */
  right?: string | number | boolean;

  /**
   * Text \`margin-bottom\`.
   */
  bottom?: string | number | boolean;

  /**
   * Text \`margin-left\`.
   */
  left?: string | number | boolean;
}
export type PrimaryComponentFoo = string | number | boolean;
export type PrimaryComponentTop = string | number | boolean;
export type PrimaryComponentRight = string | number | boolean;
export type PrimaryComponentBottom = string | number | boolean;
export type PrimaryComponentLeft = string | number | boolean;
export type PrimaryComponentSecondary = string | number | boolean;
export type PrimaryComponentSecondaryFoo = string | number | boolean;
export interface PrimaryComponentSecondarySpacing {
  /**
   * Text \`secondary\`.
   */
  secondary?: string | number | boolean;
}
export interface PrimaryComponentProps {
  boolean?: boolean;
  number?: number;

  /**
   * Text \`spacing\` or \`children\`.
   */
  spacing?: PrimaryComponentSpacing;
  foo?: PrimaryComponentFoo;

  /**
   * Text \`margin-top\`.
   */
  top?: PrimaryComponentTop;

  /**
   * Text \`margin-right\`.
   */
  right?: PrimaryComponentRight;

  /**
   * Text \`margin-bottom\`.
   */
  bottom?: PrimaryComponentBottom;

  /**
   * Text \`margin-left\`.
   */
  left?: PrimaryComponentLeft;

  /**
   * Text \`secondary\`.
   */
  secondary?: PrimaryComponentSecondary;
  secondary_foo?: PrimaryComponentSecondaryFoo;
  secondary_spacing?: PrimaryComponentSecondarySpacing;

  /**
   * Text \`spacing\` or \`children\`.
   */
  children?: React.ReactNode;
}
export default class PrimaryComponent extends React.Component<
  PrimaryComponentProps,
  any
> {
  render(): JSX.Element;
}
"
`;

exports[`generateTypes has to match SecondaryComponent snapshot 1`] = `
"import * as React from 'react';
export type ElementSecondary = string | number | boolean;

/**
 * NB: Do not change the docs (comments) in here. The docs are updated during build time by \\"generateTypes.js\\" and \\"fetchPropertiesFromDocs.js\\".
 */
export interface ElementProps extends React.HTMLProps<HTMLElement> {
  /**
   * Text \`secondary\`.
   */
  secondary?: ElementSecondary;

  /**
   * Text \`children\`.
   */
  children?: React.ReactNode;

  /**
   * SecondProperty.
   */
  secondProperty?: string;
}
declare const Element: React.FC<ElementProps>;
export default Element;
"
`;
